ereuse_devicehub/resources/device/models.py

```diff
@@ -452,6 +455,23 @@ def privacy(self):
             if privacy
         )
 
+    def add_mac_to_hid(self, components_snap=None):
+        """Returns the Naming.hid with the first mac of network adapter, 
+        following an alphabetical order.
+        """
+        self.set_hid()
+        if not self.hid:
+            return
```

Comment:
Could this code be refactored with a try except statement to make more explicit that `self.set_hid()` has failed?

Reply:
It's possible have one device without hid